testsuite/compatibility-test/src/lib.rs

```diff
@@ -31,9 +28,8 @@ fn batch_update<'t>(
     Ok(())
 }
 
-fn generate_traffic<'t>(ctx: &mut NetworkContext<'t>, validators: &[PeerId]) -> Result<()> {
+pub fn generate_traffic<'t>(ctx: &mut NetworkContext<'t>, validators: &[PeerId], duration: Duration,) -> Result<TxnStats> {
```

Comment:
If this is a function that is super useful to a number of tests maybe we should clean it up at stick it in the `forge` crate for ease of use at some point. Not needed now but just thinking about the future.

Reply:
I think ideally for `batch_upgrade` also will be widely used, we can do them both together